packages/gasket-intl/test/components.test.js

```diff
@@ -0,0 +1,152 @@
+
+    it('adds display name', function () {
+      assume(Component).property('displayName', 'withIntlProvider(MockComponent)');
+    });
+
+    describe('#render', () => {
+      it('wraps target component with IntlProvider and renders children', () => {
+        wrapper = mount(<Component { ...testProps } />);
+        assume(wrapper.find(IntlProvider)).length(1);
+        assume(wrapper.find(MockComponent)).length(1);
+      });
+
+      it('IntlProvider is passed expected default props', () => {
+        wrapper = mount(<Component { ...testProps } />);
+        const result = wrapper.find(IntlProvider);
+        assume(result.prop('locale')).eqls('en-US');
+        assume(result.prop('messages')).eqls({});
+      });
+
+      it('IntlProvider is passed expected props from Next.js page', () => {
+        testProps.pageProps = {
+          localesProps: {
+            locale: 'fr-FR',
+            messages: {
+              'fr-FR': {
+                bogus: 'BOGUS'
+              }
+            }
+          }
+        };
+        wrapper = mount(<Component { ...testProps } />);
+        const result = wrapper.find(IntlProvider);
+        assume(result.prop('locale')).eqls('fr-FR');
+        assume(result.prop('messages')).eqls({ bogus: 'BOGUS' });
+      });
+    });
+  });
+
+  describe('withLocaleRequired', function () {
+    let Component, wrapper, testProps;
+
+    beforeEach(function () {
+      Component = mod.withLocaleRequired()(MockComponent);
+      testProps = {};
+    });
+
+    it('adds display name', function () {
+      assume(Component).property('displayName', 'withLocaleRequired(MockComponent)');
+    });
+
+    describe('#render', () => {
+      it('renders null if loading', () => {
+        wrapper = mount(<Component { ...testProps } />);
+        assume(wrapper.html()).eqls(null);
+      });
+
+      it('renders custom loader if loading', () => {
+        Component = mod.withLocaleRequired('/locales', { loading: 'loading...' })(MockComponent);
+        wrapper = mount(<Component { ...testProps } />);
+        assume(wrapper.html()).eqls('loading...');
+      });
+
+      it('renders wrapped component if LOADED', () => {
+        sinon.stub(mod.helpers, 'useGasketIntl').returns(LOADED);
+        Component = mod.withLocaleRequired('/locales')(MockComponent);
+        wrapper = mount(<Component { ...testProps } />);
+        assume(wrapper.html()).includes('MockComponent');
+      });
+
+      it('renders wrapped component if ERROR', () => {
+        sinon.stub(mod.helpers, 'useGasketIntl').returns(ERROR);
+        Component = mod.withLocaleRequired('/locales')(MockComponent);
+        wrapper = mount(<Component { ...testProps } />);
+        assume(wrapper.html()).includes('MockComponent');
+      });
+    });
+  });
+
+  describe('LocaleRequired', function () {
+    let wrapper;
+
+    const doMount = props => mount(<mod.LocaleRequired { ...props }>MockComponent</mod.LocaleRequired>);
+
+    describe('#render', () => {
```

Comment:
[Passing arrow functions (aka “lambdas”) to Mocha is discouraged.](https://mochajs.org/#arrow-functions)